tempo_embeddings/embeddings/vector_database.py

```diff
@@ -48,15 +49,31 @@ def connect(self):
 
     @abstractmethod
     def insert_corpus(
-        self, collection: Collection, corpus: Corpus
+        self, collection: Union[Collection, str], corpus: Corpus
     ):
         return NotImplemented
 
     @abstractmethod
-    def retrieve_vectors_if_exist(
-        self, collection: Collection, passages: list[Passage]
+    def get_corpus(
+        self, collection: Union[Collection, str], filter_words: list[str], where_obj: dict[str, Any]
     ):
         return NotImplemented
+    
+    @staticmethod
+    def compress_embeddings(
+        corpus: Corpus,
+        umap_verbose: bool = True,
+        **umap_args,
+    ) -> ArrayLike:
+
+        if len(corpus) == 0:
+            return None
```

Comment:
I wonder if this is checked anywhere downstream. Perhaps petter raise an error to make sure the issue is noticed?
```suggestion
            raise ValueError("Empty corpus passed to compress_embeddings")
```
Alternatively, at least a warning should be logged.
```python
            logging.warning("Empty corpus passed to compress_embeddings()")
            return None
```